ci-bench/src/bench_lib/runner.rs

```diff
@@ -0,0 +1,90 @@
+use super::{Benchmark, CachegrindRunner, ReportingMode};
+use crate::bench_lib::cachegrind::InstructionCounts;
+use crate::bench_lib::BenchmarkCollection;
+use crate::cli::Side;
+use anyhow::Context;
+use itertools::Itertools;
+use rayon::iter::Either;
+use rayon::prelude::*;
+use std::collections::HashMap;
+use std::io;
+
+/// Run a single bench at the specified side
+pub fn run_single(
```

Comment:
It seems strange for this to take a `&BenchmarkCollection` + `bench_index` instead of just a single `&Benchmark`.